### ISSUE TITLE
[UPDATED] Current Development Situation

### ISSUE THREAD

**Issue by the-wright-jamie** [the-wright-jamie/tab-nine#160]
## UPDATE | 4th Apr 2024

To try and get some sort of movement and get a feature out every quarter like I said, I've opened the first development poll. Please go vote [here](https://github.com/the-wright-jamie/tab-nine/discussions/214)

<details>
<summary>
Previous Updates
</summary>

## UPDATE 2 | 29th Jan 2024

Just got merging some PRs, good stuff there and I'm going to create a release. Going forward, I'll try and keep on top of PRs. In terms of feature development from me, not sure - got some personal projects and stuff outside of programming I want to get into. And I'll continue to merge in any changes from upstream so that they actually get released to the public.

## UPDATE | 29th Jan 2024

I'm going to get looking at PRs tonight and create a release. I'm not coming back to developing features myself just yet, but I'll start addressing PRs more today. Thank you for your patience

## UPDATE | 8th Jan 2024

I won't be returning this month most likely due to unforeseen circumstances. I still intend to work on this project and I would like to name a maintainer, but unfortunately being a lone developer I don't have anyone to turn to in order to help me maintain this project. I would open up applications, but I don't think the fork is popular enough for that and being from a security background I don't feel comfortable vetting people myself for safety - if I add a maintainer and they turn out to be a bad actor, I risk my reputation. Additionally, adding maintainers would probably require some restructuring of the project such as moving it to its own organisation (as it would no longer be my fork, but a fork maintained by a 'team').

I appreciate your patience, I really hope to return within the quarter (i.e. before April) but I will have to deal with the hand that is dealt.

</details>

<details>
<summary>
Original Post
</summary>

## Original Post | 28th Nov 2023

Just a heads up: as many of us probably know, December & Christmas (and related holidays) are a busy time for everyone - including myself.

As a result, **I will be suspending active development of Tab Nine until at least the new year**.

I realise this will be disappointing to some, but I cannot neglect the obligations of the season as well as family and friends. I hope you will understand, and I look forward to resuming development in the new year.

I will still be listening for new issues and watching for new pull requests - however none will be acted upon until I'm in a better position to actually work on them. Yes, I could merge pull requests blindly, but that will just end up creating more work down the road which is the last thing I want.

In the future, most likely, we will move to a quarterly release schedule. If there is more active development, we may open a beta branch on the stores for those who want to have the newest features as soon as possible even if they are unstable. As always, you'll also be able to manually compile and install the extension on your browsers as well.

I have a very busy personal life, and as such don't have the time to spend on hobbies and interests as I once did. When I forked this project I thought that I would have more time to dedicate to it. I haven't been able to dedicate as much time as I would like to.

I'm not in a position where I am considering ceasing development - I have plenty of plans for this fork - but priorities are priorities. 
</details>

**Comment by smokedzn** [comment]
It is really unfortunate to see this fork also getting kind of inactive, although I totally understand that a browser extension doesn't really have a priority when more important things in life happen.
Are you certain that, at the latest, in Q2 we will see any new update(s)? I totally understand that all the hours of coding that would've gone into Tab9 might, at least, seem like they were for nothing, but what is your opinion on discontinuing Tab9 from your side and, rather, looking for someone who has the time and capabilities to start another fork, or rather keep this one updated? Since Tabliss hasn't received any new updates, it's pretty safe to say that Tabliss has reached its end of development, thus making your fork the preferred “version” between both of them.
In my personal opinion, it would be a significant decision to either try to keep Tab9 from Q2 onwards updated or look for someone else who is willing to try to keep this fork or another (new) fork updated.

Of course, I'm not saying that you should discontinue the development of Tab9 in any way, but rather, am I asking how the future of Tab9 will actually look like and if we can actually expect updates from Q2 on.
Unfortunately, I don't really have much experience with coding, which is why I can't really help that much directly, but rather only with small additions, ideas, or translations.

Anyway, I would be grateful to hear an opinion on my questions and hope that everything that stands in the way of development and further can be resolved quickly! :)

**Comment by the-wright-jamie** [comment]
# RE: https://github.com/the-wright-jamie/tab-nine/issues/160#issuecomment-1883842685

> It is really unfortunate to see this fork also getting kind of inactive

Yeah, it is, this was my mistake for underestimating the load required for maintaining an open source project 😅.

> Are you certain that, at the latest, in Q2 we will see any new update(s)? 

I can't be *certain*, but there's a good amount of stuff in the backlog to at least achive my goal of trying to roll an update out every Q.

> opinion on discontinuing Tab9 from your side and, rather, looking for someone who has the time and capabilities to start another fork, or rather keep this one updated?

Preferably, I would create a GitHub organization and keep Tab Nine alive with some maintainers - not much point in creating another fork when I am still here and able to respond to people, just not able to spend as long as I would like. I need to fix some DevOps stuff with the automated deployment but once that's done I (or maintainers) would just need to merge changes and then create releases greatly reducing the release workload. 

> it would be a significant decision to either try to keep Tab9 from Q2 onwards updated or look for someone else who is willing to try to keep this fork or another (new) fork updated.

Yeah, I can sort of feel the weight of this. I originally created this fork in order to fix the GitHub Calendar issues and bring in some features I wanted - specifically the dark mode and accent colours. I thought it would be a good idea to let people know I had decided to publish my fork but I didn't expect anyone to actually pay attention! 

It all really depends on how much community help comes as I don't see my personal workload reducing (I would also like to learn Rust, which I can't do while also trying to maintain Tab9 every night). Health, work, other issues are all getting in the way. I would love to be able to finish work at the end of the day and just hop onto Tab9/other coding stuff, but I also have a bunch of other passions and hobbies.

Basically, I should have considered how much was actually involved in a fork before starting. I think one update a Q is sustainable though, I kind of set a fast development track record in the first few months and I held myself to that standard. Need to get a proper release schedule to make this sustainable and I think an update a Q is better than more community fracturing in a new fork or no more updates ever. I hope this is a desirable outcome for you 🙂

**Comment by the-wright-jamie** [comment]
## Update
I'm going to get looking at PRs tonight and create a release. I'm not coming back to developing features myself just yet, but I'll start addressing PRs more today. Thank you for your patience

**Comment by the-wright-jamie** [comment]
Going to take it as it comes, let's see how this goes

**Comment by the-wright-jamie** [comment]
## UPDATE | 4th Apr 2024
To try and get some sort of movement and get a feature out every quarter like I said, I've opened the first development poll. Please go vote [here](https://github.com/the-wright-jamie/tab-nine/discussions/214)
